### PR TITLE
Companion PR to substrate Rename StorageMap::exists to ::contains_key

### PR DESCRIPTION
**Pull request by apopiak**
fixes usages of `StorageMap::exists(key)` by using the new `StorageMap::contains_key(key)` introduced in paritytech/substrate#4847 instead.
See Issue paritytech/substrate#4839 for the motivation.

### CODE REVIEW

**Comment by apopiak** [comment]
@bkchr Not sure what's up with the
```
error: the lock file /builds/parity/polkadot/Cargo.lock needs to be updated but --locked was passed to prevent this
```
message.

**Comment by NikVolf** [comment]
@apopiak run `cargo test --all` (no need to wait it is finished) and then commit Cargo.lock

**Comment by apopiak** [comment]
Also unsure how this import could not resolve :sweat_smile: 
```
 error[E0432]: unresolved import `frame_support::traits::VestingCurrency`
```

**Comment by bkchr** [comment]
https://github.com/paritytech/polkadot/pull/816 this needs to be merged first.

File: availability-store/Cargo.toml [hunk 1]
```diff
@@ -15,15 +15,15 @@ futures = "0.3.4"
...
-sc-network = { git = "https://github.com/paritytech/substrate", branch = "polkadot-master" }
-consensus_common = { package = "sp-consensus", git = "https://github.com/paritytech/substrate", branch = "polkadot-master" }
-client = { package = "sc-client-api", git = "https://github.com/paritytech/substrate", branch = "polkadot-master" }
-sc-client = { git = "https://github.com/paritytech/substrate", branch = "polkadot-master" }
-sp-runtime = { git = "https://github.com/paritytech/substrate", branch = "polkadot-master" }
-sp-blockchain = { git = "https://github.com/paritytech/substrate", branch = "polkadot-master" }
-sp-api = { git = "https://github.com/paritytech/substrate", branch = "polkadot-master" }
-keystore = { package = "sc-keystore", git = "https://github.com/paritytech/substrate", branch = "polkadot-master" }
-sp-core = { git = "https://github.com/paritytech/substrate", branch = "polkadot-master" }
+sc-network = { git = "https://github.com/paritytech/substrate", branch = "master" }
```
Comment (bkchr) [inline availability-store/Cargo.toml]: Please revert.

We always point to the `polkadot-master` branch. You can try to update Substrate or just let the pr sit here until somebody else updates Substrate and pulls in your pr.

Comment (apopiak) [reply availability-store/Cargo.toml]: Right, makes sense.
The PR was just not building with my substrate branch (which was already merged into master), so I was unsure how to check whether everything works out and tried using substrate `master`.
So do I switch it to `polkadot-master` and then just wait for someone to update it?

Comment (apopiak) [reply availability-store/Cargo.toml]: I guess my question is: How do I verify that my PR is still valid when we merge it?

Comment (bkchr) [reply availability-store/Cargo.toml]: The pr will be merged into another one, that runs CI before. So, we will not merge this one directly. If you want to have this merged, you need to update the `polkadot-master` branch in Substrate and fix all the compile errors in Polkadot.

**Comment by bkchr** [comment]
Superseded by https://github.com/paritytech/polkadot/pull/834

**Pull request closed without merging**